### PR TITLE
move pages tab tabs to canjs and improve UX

### PR DESCRIPTION
**Pull request by janebitovi**
also update references to vars when they're renamed

Change log:
**Pages -> Page Info section**
- no visual or UX change, using CanJS now

**Pages -> Question Info tab/section**
Now uses CanJS plus new UX improvements

Old:
image
New:
image

The file upload field has been improved #224 
Similar UX to the Var Picker introduced on the Buttons tab previously:
image

Can search existing files (which should be filtered by extension as a future improvement, but for now shows all media)
just like we search vars on the buttons tab. Upload button does the same and updates the list with the new file. Also now shows when the string entered does not match an existing file.
image

Counting var and outer loop var both now use the improved Var Picker from the Buttons tab
image

**Pages -> Learn more**
Now uses CanJS plus UX improvements for the 3 file picking fields:
image
using the same File Picker component from the Questions tab

**Pages -> Fields tab**
Now uses CanJS plus UX improvements introduced on the buttons tab
Buttons instead of drag drop: 
old
image
new
image

The var picker and file picker components have been used here as well
image

The file picker for the "External List" field lists all xml files already available in the guide, future improvement would be to filter out Guide.xml from this list
image

Fields that expect numbers now auto-correct themselves on blur to strip problematic characters:
numbers-only

bugfix: If a type can't be required (such as user avatar), the required field now becomes false when switching to the type that doesn't allow it. Previously the un-required-able field could be locked into being required if required was checked and the type changed to something like User Avatar.

**Pages -> Buttons tab**
bugfix: the add button would sometimes add two buttons instead of just one, that has been fixed

**Pages -> Adv Logic**
No changes yet, this is still the legacy section

----

**Author Variables tab**

related to #225 

Find usage was using the name as it was entered on the form instead of whatever it's actually currently set to. This has been fixed and the UI now indicates the var name that will be searched for when the find usage button is clicked:
old:
old-var
new:
new-var

further, and most importantly, renaming a variable and pressing save now updates all references to that variable so no logic etc breaks when renaming vars:
rename-var

### CODE REVIEW

**Comment by janebitovi** [comment]
in progress

add "edit page" button to save the current page and reopen the editor for the next page for #226 
image

add "add new" button to every var field so authors can open the variable dialog and add a new one while working from the page editor for #137 
image

**Comment by janebitovi** [comment]
new variables can now be added from every/any var field in the qde page editor
image

new-var-from-qde

buttons with a 'next' destination set now also show an "edit page" button that when clicked, switches the QDE to the editor for that destination page.
edit-next

**Comment by janebitovi** [comment]
Moved Adv Logic tab to CanJS + UX improvements
and took care of CCALI/CAJA#164, #146, and #200 

goto-check-advanced-logic
quotes-advanced-logic
advanced-logic

File: CAJA_WS.php [hunk 1]
```diff
@@ -213,6 +216,27 @@
...
+
+				$xmlList[]=(object)[
+					"gid" => $gid,
+					"ownerfolder" => $ownerfolder,
+					"name" => $xmlName,
+					"filename" => $filename,
+					"rename" => "",
+					"extension" => $extension,
+					"modified" => date (DATE_FORMAT_UI, filemtime($xmlFilePath)),
+					// "path" => $xmlFilePath,
```
Comment (mitch-mitchel) [inline CAJA_WS.php:235]: should this comment be removed, or do we maybe need in the future?

File: src/pages-tab/components/advanced-logic/advanced-logic.stache [hunk 2]
```diff
@@ -1,3 +1,11 @@
 <can-import from="./advanced-logic.less" />
+<can-import from="./logic-editor/" />
 
-{{{legacySection}}}
+<fieldset name="record">
+  <legend>Advanced Logic</legend>
+  <logic-editor label:from="'Before'" key:from="'codeBefore'" page:from="page" appState:from="appState" />
```
Comment (mitch-mitchel) [inline src/pages-tab/components/advanced-logic/advanced-logic.stache]: this is just a question, i've used `label:raw="Before"` in other places, without the nested single/double quotes. they are just 2 syntaxes for the same end result, right?

File: src/pages-tab/components/advanced-logic/logic-editor/logic-editor.js [hunk 3]
```diff
@@ -0,0 +1,277 @@
+// import $ from 'jquery'
+import DefineMap from 'can-define/map/map'
+import Component from 'can-component'
+import template from './logic-editor.stache'
+import _unescape from 'lodash/unescape'
+
+export const LogicEditor = DefineMap.extend('LogicEditor', {
```
Comment (mitch-mitchel) [inline src/pages-tab/components/advanced-logic/logic-editor/logic-editor.js]: I've added VM to both of these names in other places, but I feel like this is more of a style choice. If you prefer it without, the code should probably standardize on one style or the other

my style:
export const LogicEditorVM = DefineMap.extend('LogicEditorVM', {

File: src/pages-tab/components/advanced-logic/logic-editor/logic-editor.stache [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<can-import from="./logic-editor.less" />
+
+{{<listUI}}
```
Comment (mitch-mitchel) [inline src/pages-tab/components/advanced-logic/logic-editor/logic-editor.stache:3]: i forgot about these inline partials - awesome

File: src/pages-tab/components/var-picker/field/var-picker-field.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
+// import $ from 'jquery'
```
Comment (mitch-mitchel) [inline src/pages-tab/components/var-picker/field/var-picker-field.js]: can remove?

**Review by mitch-mitchel — APPROVED** [review]
This awesome!

**Pull request merged**